bika/lims/content/attachment.py

```diff
@@ -177,33 +185,30 @@ def getAnalysis(self):
         """Return the analysis to which this is linked it may not be linked to
         an analysis
         """
-        tool = getToolByName(self, REFERENCE_CATALOG)
-        uids = [uid for uid in
-                tool.getBackReferences(self, 'AnalysisAttachment')]
-        if len(uids) == 1:
-            reference = uids[0]
-            analysis = tool.lookupObject(reference.sourceUID)
-            return analysis
-        return None
+        analysis = get_backreferences(self, 'AnalysisAttachment',
+                                      as_brains=True)
+        if not analysis:
+            return None
+
+        if len(analysis) > 1:
+            logger.warn("Single attachment assigned to more than one Analysis")
```

Comment:
Just a question for better understanding: Why is it not allowed to have a single attachment attached to more than one Analysis? Wouldn't it make sense?

Reply:
Nothing stops you to add more than one Attachment to the same Analysis, of course. Even the same file. What you cannot do is to associate the same `Attachment` object (with the same UID) to more than one Analysis. Every time you add an attachment, a new `Attachment` object will be created and associated to an Analysis, even if that Analysis already has other `Attachment` objects associated. Here we are using `self`(this particular instance of `Attachment`, with it's own UID) to get its back references, that atm, can only be one Analysis.
We could change and allow the user to associate multiple analyses to a single analysis in just one shot, of course. Then, the first thing we'd need to change is the attachments widget.